pkg/asset/manifests/infrastructure.go

```diff
@@ -0,0 +1,98 @@
+package manifests
+
+import (
+	"path/filepath"
+
+	"github.com/ghodss/yaml"
+	"github.com/pkg/errors"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/asset/templates/content/openshift"
+
+	configv1 "github.com/openshift/api/config/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	infraCrdFilename = filepath.Join(manifestDir, "cluster-infrastructure-01-crd.yaml")
+	infraCfgFilename = filepath.Join(manifestDir, "cluster-infrastructure-02-config.yml")
+)
+
+// Infrastructure generates the cluster-infrastructure-*.yml files.
+type Infrastructure struct {
+	FileList []*asset.File
+}
+
+var _ asset.WritableAsset = (*Infrastructure)(nil)
+
+// Name returns a human friendly name for the asset.
+func (*Infrastructure) Name() string {
+	return "Infrastructure Config"
+}
+
+// Dependencies returns all of the dependencies directly needed to generate
+// the asset.
+func (*Infrastructure) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.InstallConfig{},
+		&openshift.InfrastructureCRD{},
+	}
+}
+
+// Generate generates the Infrastructure config and its CRD.
+func (i *Infrastructure) Generate(dependencies asset.Parents) error {
+	installConfig := &installconfig.InstallConfig{}
+	infra := &openshift.InfrastructureCRD{}
+	dependencies.Get(installConfig, infra)
+
+	var platform configv1.PlatformType
+	switch p := installConfig.Config.Platform; {
+	case p.AWS != nil:
+		platform = configv1.AWSPlatform
```

Comment:
do you think we can set the other platforms too ? like libvirt/openstack

also you can switch on `platform.Name()`

Reply:
Done